WordPress/gutenberg
# Scripts: Replace url-loader for SVGs with asset modules

## What?
This PRs tries to replace `url-loader` for SVGs with [Asset Modules](https://webpack.js.org/guides/asset-modules/) which were introduced in webpack 5.

## Why?
The current solution is deprecated and it's recommend [in the documentation](https://react-svgr.com/docs/migrate/#webpack) of `@svgr/webpack` to use the asset modules. Also, If you want to use a SVG as a React component you currently have to explicitly import a `ReactComponent` like so:

    import { ReactComponent as Star } from './star.svg';


## How?
I'm removing `url-loader` as a dependency and updated the implementation based on https://react-svgr.com/docs/webpack/#use-svgr-and-asset-svg-in-the-same-project. Now you can do:

    import Star from './star.svg'; // React component
    import starUrl from './star.svg?url'; // Asset URL


Unfortunately, `'./star.svg?url'` is currently causing an ESLint error:

    Unable to resolve path to module './star.svg?url'. eslint import/no-unresolved

That's why I added the [webpack resolver](https://www.npmjs.com/package/eslint-import-resolver-webpack). But I'm currently not sure if that's a good idea and/or whether this can be part of the eslint-plugin instead and maybe a simple custom resolver which only handles [this part](https://github.com/import-js/eslint-plugin-import/blob/747d6dc5b905a718c89f10aad04cb2bcdbed7068/resolvers/webpack/index.js#L46-L50) should be created.

## Testing Instructions
1. Run `npx wp-create-block example-static --no-wp-scripts`
2. Download [logo.svg](https://user-images.githubusercontent.com/617637/158021074-3654416f-c360-4dfa-9ab0-702b0230efe5.svg) and put it into `example-static`
3. Add imports to `edit.js`
```
import WordPressLogo from './logo.svg';
import wordPressLogo from './logo.svg?url';
```
4. Update `Edit()` by adding the following into the existing paragraph element:
```
<WordPressLogo />
<img src={ wordPressLogo } alt={ __( 'WordPress Logo' ) } />
```
5. Run `../node_modules/.bin/wp-scripts build`

The build folder should now have an `images` directory with the SVG and `index.js` has also the SVG element.

Visual output:
<img width="981" alt="" src="https://user-images.githubusercontent.com/617637/158021195-1fa7a6b9-a73e-4c82-9aa8-3178952b55a5.png">
